nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -202,7 +210,22 @@ class SimpleActionServer
 
       // Return quickly to avoid blocking the executor, so spin up a new thread
       debug_msg("Executing goal asynchronously.");
-      execution_future_ = std::async(std::launch::async, [this]() {work();});
+      execution_future_ = std::async(std::launch::async, [this]() {
```

Comment:
Consider extracting added code to function like: setSoftRealTimePriority() and call it here. It would improve readability.